debian/vpinstall.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH VPINSTALL 1 "November 19, 2009"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
vpinstall \- script to configure an attached Vantage Pro for wview installation
.SH SYNOPSIS
.B vpinstall
.RI
.br
.SH DESCRIPTION
This manual page describes the use of
.B vpinstall
.
Installation configuration of a wview Vantage Pro installation.
Utilizes the vpconfig utility to configure the station.
.BR
.BR Note: must be executed on the wview server.
.SH OPTIONS
There are no options.
.TP
.SH SEE ALSO
.BR vpconfig (1),
.BR /usr/share/doc/wview/wview-User-Manual.html
.br
.SH AUTHOR
vpinstall was written by Mark Teel.
.PP
This manual page was written by Mark Teel <mteel2005@gmail.com>,
for the Debian project (and may be used by others).


ChangeLog
RELEASE VERSION HISTORY (newest to oldest)
------------------------------------------


5.21.5    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Remove debug message.




5.21.4    04-06-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Added stale pid file cleanup to the APT package start script.

2)  WMRUSB HID interface changes. Changed reader thread to blocking IO.
    Mitigated the console occasionally blasting out bogus data. Now zero
    packet loss. Fixed issue with pressure sanity check. Added sanity
    check for bogus rain accumulator values from the WMR console.




5.21.3    03-31-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Further WMRUSB interface improvements. Better packet framing.




5.21.2    03-30-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Updated debian APT postinst script to include copying the exfoliation skin
    for new installs.

2)  Enabled checksum processing for WMR stations.

3)  Changed LOOP windspeed items to floats for better precision.
    NOTE: If you use custom datafeed clients the LOOP data structure has changed.

4)  Added temp only packet support for WMRUSB stations. This adds support for
    add-on sensors such as the pool sensor.

5)  Fixed APT installs changing the wview.log file ownership incorrectly. Also
    restarts rsyslog automatically if a new install.




5.21.1    03-13-2014
--------------------

radlib version required: 2.12.0 or newer

1)  Modified HILOW database storage so higher precision is used for cumulative
    data such as rain and ET.

2)  Fixed issue with yearly historical charts when generated at wview start
    (as opposed to at midnight). The current day was erroneously being included in
    the history data. This was most apparent with the UV and Solar Radiation
    charts.

3)  Fixed bug introduced in 5.21.0 release causing NULL wind direction to have
    360 added to it (to normalize calibration result). A value of -99640 was
    being stored/displayed.




5.21.0    03-05-2014
--------------------

radlib version required: 2.12.0 or newer

1)  WMRUSB improvements. Further reverse engineering of the undocumented protocol. Reset and
    heartbeat messages now working. Packet reception more stable. Initial packet acquisition
    faster resulting in faster startup.

2)  Modify CWOP packet header for new stricter compliance requirements of CWOP servers.

3)  Fix log message time_t bug for 64-bit platforms.

4)  Add timeout parameter for ssh rules. Added rows to config database.

5)  Updated data feed conversion logic to handle ARCHIVE_VALUE_NULL in loop data.

6)  Added flat line sensor value detection for outside temp, wind direction and wind speed.
    If there are no changes over a 4 hour period an email alert will be sent.

7)  Added Station Name, City and State to the wviewmgmt System Status page header.

8)  Added logic to SSH timer to fixup next timer interval if system time has skewed or
    if the ssh transfer took longer than 2 minutes.

9)  Fixed TE923 rain calculation to assume each tip is 1/36" instead of 1mm.

10) Fixed a bug for Vantage Pro stations which could cause the last archive interval of
    the previous day to be included in the current day HILOW data. This occurred if the
    midnight archive record was provided late by the console.

11) Added wet bulb temperature calculation and html tag --WetBulbTemp--. Added to all
    "readings.incx" templates.

12) Added email alert notification if no loop data has been received for a given archive
    interval. This applies to all stations except the Davis stations (alerts already
    exist for them).

13) Updated start scripts to detect and delete stale pid files.

14) Modified WH1080 data parsing to assume a humidity of 9% if the value from the
    station is undefined (0xff).




5.20.2    03-21-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added back "dailyrainin" for http reporting. They do no computation based on hourly rain values.

2)  Fixed a typo in the wviewmgmt station.php file. 




5.20.1    03-20-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Modified http rain submission (Wunderground and Weatherforyou) so only the
    "rainin" parameter is sent so daily rain will be correct. 




5.20.0    03-17-2012
--------------------

radlib version required: 2.12.0 or newer

1)  Added "wait_for_time_set" function to most wview start script examples so 
    wview daemons are not started until the system time is set. This applies to
    platforms without a real time clock which must reset their time during 
    startup.

2)  Updated the User Manual section on skin selection to better describe enabling
    extended data generation (UV, ET, radiation).

3)  Updated the debian post install script to try harder to find the default http
    document root location for creation of symbolic links to wviewmgmt and the
    weather home page.

4)  Added new configuration option to optionally disable using the Vantage Pro 
    console archive records and instead autogenerate them from the LOOP data.
    This is how all other stations work. It also supports non-standard console
    interfaces (those not using the datalogger on the console).

5)  Updated the Weatherforyou URL for data submissions.

6)  Fixed bug with DataFeed clients that caused an infinite loop if you
    restarted wview while there was a datafeed client connection.

7)  Fixed bug with Alarm Type configuration introduced in 5.19.0.

8)  Changed TWI station to compute rain fall rate as it is reported to not
    be provided by the station.

9)  Added socket retry logic to the station ethernet interface (used by the
    Vantage Pro IP datalogger and any serial-to-ethernet device).

10) Modified CWOP host name to IP address resolution so that 'getaddrinfo' is
    used instead of 'gethostbyname' and all results are tried in the order they 

stations/VantagePro/vpconfig/vpinstall.sh
###########################################################################
#
# File:           vpinstall.sh
#
# Description:    Provide a script to interactively configure a new
#                 Vantage Pro installation.
#
# Usage:          vpinstall
#
# History:
# Engineer	  Date	    Ver   Comments
# MS Teel	  04/14/05   1    Original
#
###########################################################################
#
#
################### MACROS ######################
#
SCRIPTNAME=vpinstall
SCRIPTNAME_VERSION="1.0.1"

WVIEWD_PID=/var/wview/wviewd.pid
VPCONFIG_BIN=$WVIEW_INSTALL_DIR/vpconfig
test -x $VPCONFIG_BIN || (echo "vpconfig not installed!"; exit 2)

# default values
DEVNAME=/dev/ttyS0
ARCINT=5
ELEVATION=0
LATITUDE=0
LONGITUDE=0
RAINSTART=1
RAINTODATE=0
ETTODATE=0

################## INCLUDE ######################
#

############### HELP FUNCTIONS ##################
#

#################### MAIN #######################
#

# First test to make sure wview is not running...
if [ -f $WVIEWD_PID ]; then
    echo "wviewd is running - stop wview before configuring the VP console"
    exit 3
fi

echo "################################################################################"
echo ""
echo "--> Installation Configuration for Vantage Pro(2) Console"
echo ""
echo "--> To reset historical highs and lows, use the Vantage Pro Console"
echo ""
echo "--> Note: values in parenthesis are used if you hit enter at a prompt..."
echo ""
echo "################################################################################"
echo ""

echo ""
echo "-------------------------------------------------------------"
echo "--NOTE-- ** Serial and USB data loggers are both considered serial -"
echo "--NOTE--    examples of this are: /dev/ttyS0, /dev/ttyUSB0, /dev/cuaa0, etc."
echo "--NOTE-- ** If your Vantage Pro is connected to a terminal or serial server,"
echo "--NOTE--    you should enter the host and port in the format host:port -"
echo "--NOTE--    examples of this are: 10.10.10.10:3001, xyplex1:2102, etc."
echo "Serial port or host:port the VP console is connected to (i.e. /dev/ttyS0, mss1:3001)"
echo -n "(/dev/ttyS0): "
read INVAL
if [ "" != "$INVAL" ]; then
    DEVNAME=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Desired archive interval (minutes) (5,10,15,30,60)"
echo -n "(5): "
read INVAL
if [ "" != "$INVAL" ]; then
    ARCINT=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Station elevation (feet) ((-2000) - (15000))"
echo -n "(0): " 
read INVAL
if [ "" != "$INVAL" ]; then
    ELEVATION=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Station latitude (tenths of degrees) ((-900) - (900), (-) => South)"
echo -n "(0): "
read INVAL
if [ "" != "$INVAL" ]; then
    LATITUDE=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Station longitude (tenths of degrees) ((-1799) - (1800), (-) => West)"
echo -n "(0): "
read INVAL
if [ "" != "$INVAL" ]; then
    LONGITUDE=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Rain season start (month) (1 - 12)"
echo -n "(1): "
read INVAL
if [ "" != "$INVAL" ]; then
    RAINSTART=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Year-to-Date Rain (inches) (0.00 - 200.00)"
echo -n "(0.00): "
read INVAL
if [ "" != "$INVAL" ]; then
    RAINTODATE=$INVAL
fi

echo ""
echo "-------------------------------------------------------------" 
echo "Year-to-Date ET (inches) (0.00 - 200.00)"
echo -n "(0.00): "
read INVAL
if [ "" != "$INVAL" ]; then
    ETTODATE=$INVAL
fi

echo ""
echo ""
echo "-------------------------------------------------------------" 
echo "Let's review your answers:"
echo ""
echo "VP Console Device:         $DEVNAME"
echo "Archive Interval:          $ARCINT (minutes)"
echo "Station Elevation:         $ELEVATION (feet)"
echo "Station Latitude:          $LATITUDE (tenths of a degree)"
echo "Station Longitude:         $LONGITUDE (tenths of a degree)"
echo "Rain Season Start Month:   $RAINSTART"
echo "Year-to-Date Rain:         $RAINTODATE"
echo "Year-to-Date ET:           $ETTODATE"
echo "-------------------------------------------------------------" 
echo ""
echo "WARNING: All archive records will be erased on the VP console!"
echo "Do you want to proceed with these values? (y/n)"
echo -n "(n): "
read INVAL
if [ "y" = "$INVAL" ]; then
    echo ""
    echo "Configuring with new values (takes some time, be patient) ..."
    echo ""
else
    echo ""
    echo "Aborting Vantage Pro Configuration ..."
    exit 4
fi

# OK, here we go ...

# 1) Set the elevation
$VPCONFIG_BIN $DEVNAME setelevation $ELEVATION
if [ "$?" != "0" ]; then
    echo "set elevation failed!"
    exit 7;
fi

sleep 1

# 2) Set the rain season start
$VPCONFIG_BIN $DEVNAME setrainseasonstart $RAINSTART
if [ "$?" != "0" ]; then
    echo "set rain season start failed!"
    exit 7
fi

sleep 1

# 3) Set the rain year-to-date
$VPCONFIG_BIN $DEVNAME setyeartodaterain $RAINTODATE
if [ "$?" != "0" ]; then
    echo "set year-to-date rain failed!"
    exit 7
fi

sleep 1

# 4) Set the ET year-to-date
$VPCONFIG_BIN $DEVNAME setyeartodateET $ETTODATE
if [ "$?" != "0" ]; then
    echo "set year-to-date ET failed!"
    exit 7
fi

sleep 1

# 5) Set the archive interval
$VPCONFIG_BIN $DEVNAME setinterval $ARCINT
if [ "$?" != "0" ]; then
    echo "set Archive Interval failed!"
    exit 7
fi

sleep 5

# 6) Set the LAT/LONG
$VPCONFIG_BIN $DEVNAME setlatlong $LATITUDE $LONGITUDE
if [ "$?" != "0" ]; then
    echo "set LAT/LONG failed!"
    exit 7
fi


echo ""
echo -n "Please wait - allowing VP console to digest new settings "
sleep 2
echo -n "."
sleep 2
echo -n "."
sleep 2
echo -n "."
sleep 2
echo -n "."
sleep 2
echo -n "."
echo ""
echo ""

$VPCONFIG_BIN $DEVNAME show

echo ""
echo "################################################################################"
echo "It is advisable to wait $ARCINT minutes before starting wview for the" 
echo "first time so the Vantage Pro can become consistent with the new settings..."
echo ""
echo "In the meantime, be sure you have deleted any archive files in"
echo "/var/wview/archive if you have changed the archive interval..."
echo ""
echo "Vantage Pro configuration complete!"
echo ""
echo "################################################################################"
exit 0

